tests/rbd/rbd_tier0.py

```diff
@@ -0,0 +1,299 @@
+            return 1
+
+        # Display image info and verify
+        image_info = rbd.image_info(pool, image)
+        log.info(f"Image info : {image_info}")
+        if image not in image_info["name"]:
+            log.error(f"Image {image} in not listed for pool {pool}")
+            return 1
+
+        # List image metadata, set last_update metadata and remove it and verify
+        image_meta = rbd.image_meta(action="list", image_spec=image_spec)
+        try:
+            image_meta = json.loads(image_meta)
+            log.info(f"Image meta : {image_meta}")
+        except ValueError as e:
+            log.error(f"Image meta fetch failed : {image_meta} with error {e}")
+
+        # verify image info and log any errors
+
+        # Set last_update metadata and remove it and verify both
+        date = datetime.today().strftime("%Y-%m-%d")
+        rbd.image_meta(
+            action="set",
+            image_spec=image_spec,
+            key="last_update",
+            value=date,
+        )
+
+        update_date = rbd.image_meta(
+            action="get", image_spec=image_spec, key="last_update"
+        )
+        log.info(f"Updated date : {update_date}")
+        if update_date.strip() != date:
+            log.error("Setting image metadata last_update failed")
+            return 1
+
+        rbd.image_meta(
+            action="remove",
+            image_spec=image_spec,
+            key="last_update",
+        )
+        update_date = rbd.image_meta(
+            action="get", image_spec=image_spec, key="last_update"
+        )
+        log.info(f"Updated date : {update_date}")
+        if update_date != 1:
+            log.error("Removing image metadata last_update failed")
+            return 1
+
+        # Create snapshot, protect, clone, flatten, resize, remove, rollback
+        if rbd.snap_create(pool, image, snap) != 0:
+            log.error(f"Snapshot create failed for {snap_name}")
+            return 1
+
+        if rbd.protect_snapshot(snap_name) != 0:
+            log.error(f"Snapshot protect failed for {snap_name}")
+            return 1
+
+        if rbd.create_clone(snap_name, pool, clone) != 0:
+            log.error(f"Clone create failed for {clone}")
+            return 1
+
+        if rbd.flatten_clone(pool, clone) != 0:
+            log.error(f"Clone flatten failed for {clone}")
+            return 1
+
+        resize_int = size_int + 2
+        resize = f"{resize_int}{size.replace(str(size_int),'')}"
+        if rbd.image_resize(pool, clone, resize) != 0:
+            log.error(f"Clone resize failed for {clone}")
+            return 1
+
+        rbd_info = rbd.image_info(pool_name=pool, image_name=clone)
+        if int(rbd_info["size"] / (1024 * 1024 * 1024)) != resize_int:
+            log.error(f"Image resize failed on clone {clone}")
+            return 1
+
+        # verify rollback functionality, resize original image, rollback to the snap
```

Comment:
It would be nice to add a comment where that method exist.

Reply:
Which method Sunil?

Reply:
I mean in the down each operation you are doing there if a comment exist would be fine I thought anyway it's small statement it's ok to here as well, leave it.